hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/entity/AccountBuilderImpl.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.token.impl.entity;
+
+import com.hedera.node.app.service.token.entity.Account;
+import com.hedera.node.app.service.token.entity.AccountBuilder;
+import com.hedera.node.app.spi.key.HederaKey;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An implementation of {@link AccountBuilder} for building Account instances. This class is
+ * <strong>not</strong> exported from the module.
+ */
+public class AccountBuilderImpl implements AccountBuilder {
+    private final Account copyOf;
+
+    // These fields are the ones that can be set in the builder
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    private Optional<HederaKey> key;
```

Comment:
Do you have in mind replacing the empty `KeyList` we use for e.g. `0.0.800` with an empty `Optional`?

Reply:
Yes, we can delete that. Will add a comment.